includes/utilities.inc

```diff
@@ -183,6 +183,11 @@ function islandora_basic_collection_get_other_parents(AbstractObject $object, Ab
  *   The collection object to add to.
  */
 function islandora_basic_collection_add_to_collection(AbstractObject $new_member, AbstractObject $collection) {
+  // Create hook_islandora_basic_collection_share_migrate()
+  foreach (module_implements('islandora_basic_collection_share_migrate') as $module) {
+    $function = $module . '_islandora_basic_collection_share_migrate';
+    $function($new_member, $collection);
+  }
   $results = $new_member->relationships->get(FEDORA_RELS_EXT_URI, 'isMemberOfCollection', $collection->id);
   if (empty($results)) {
```

Comment:
Also I think you might want your hook invocation inside this `if` statement. Otherwise it will trigger even if the object is already a member of the collection.

Reply:
Hmm, I'm running into this problem even with the invoke inside the if statement. I don't understand why, though.

Reply:
Turns out the undesired firigin was because of the "fake:pid" used throughout the module. I don't know what it's for, but excluding it fixes the undesired firing.